filebeat/docs/howto/migrate-to-filestream.asciidoc

```diff
@@ -0,0 +1,243 @@
+[[migrate-to-filestream]]
+== Migrate log input configurations to filestream
+
+Filestream input is generally available since 7.14. So it is high time you
+migrate your existing log input configurations.  It comes with many
+improvements over the old input like configurable order for parsers and more.
+
+While we do not plan to remove the log input from Filebeat, we are not fixing
+new issues or adding any enhancements to the input. Our focus is on filestream.
+
+In this guide, we are leading you through the migration of an existing log input configuration.
+Let's say you are collecting the list of files above with the following three log inputs:
+
+[source,yaml]
+----
+filebeat.inputs:
+ - type: log
+   enabled: true
+   paths:
+     - /var/log/java-exceptions*.log
+   multiline:
+    pattern: '^\['
+    negate: true
+    match: after
+  close_removed: true
+  close_renamed: true
+
+- type: log
+  enabled: true
+  paths:
+    - /var/log/my-application*.json
+  scan_frequency: 1m
+  json.keys_under_root: true
+
+- type: log
+  enabled: true
+  paths:
+    - /var/log/my-old-files*.log
+  tail_files: true
+----
+
+The list of the collected files and the progress of data collection with the log input:
+["source","sh",subs="attributes"]
+----
+/var/log/java-exceptions1.log (100%)
+/var/log/java-exceptions2.log (100%)
+/var/log/java-exceptions3.log (75%)
+/var/log/java-exceptions4.log (0%)
+/var/log/java-exceptions5.log (0%)
+/var/log/my-application1.json (100%)
+/var/log/my-application2.json (5%)
+/var/log/my-application3.json (0%)
+/var/log/my-old-files1.json (0%)
+----
+
+=== 1. Set an identifier for each filestream input
+
+All filestream inputs require an ID. So make sure you set a unique identifier for every input. 
+
+Warning: Never change the ID of an input otherwise you will end up with duplicate events.
+
+[source,yaml]
+----
+filebeat.inputs:
+- type: filestream
+  enabled: true
+  id: my-java-collector
+  paths:
+    - /var/log/java-exceptions*.log
+
+- type: filestream
+  enabled: true
+  id: my-application-input
+  paths:
+    - /var/log/my-application*.json
+
+- type: filestream
+  enabled: true
+  id: my-old-files
+  paths:
+    - /var/log/my-old-files*.log
+----
+
+=== 2. Exclude all processed files
+
+Filebeat does not provide access to the state information of different inputs.
+Hence, filestream cannot access the state information of a log input in the
+registry of Filebeat. You have to exclude the files the log input has processed
+or is processing. If you do not exclude those files, you will end up with
+duplicate events in the output.
+
+Given the file list above with their ingestion progress, you
+should run log input and filestream input simultaneously until everything
+collected by the log input has made it to the output.
+Once the files collected by the log input are shipped, you can delete log
+inputs and the exlude_files settings from filestream input.
```

Comment:
```suggestion
Given the file list and ingestion progress shown earlier, 
you should run the `log` and `filestream` inputs simultaneously until everything
collected by the `log` input has made it to the output.
After the files collected by the `log` input are shipped, you can delete the `log`
inputs and the `exlude_files` settings from `filestream` input.
```

Reply:
I wonder if this is the right place to mention this? It makes it sound like the users should run Filebeat right now, but I think you want them to wait until the configuration is complete, no? It might be worth moving this to a fourth section. Something like, Step 4: Run Filebeat with `log` and `filestream` inputs enabled